peachjam/adapters/adapters.py

```diff
@@ -158,10 +179,78 @@ def update_document(self, url):
             # for stub documents, use the publication document as the source file
             pubdoc = document["publication_document"]
             if pubdoc and pubdoc["url"]:
-                self.download_source_file(pubdoc["url"], doc, title)
+                self.download_source_file(pubdoc["url"], created_doc, title)
         else:
             # the source file is the PDF version
-            self.download_source_file(f"{url}.pdf", doc, title)
+            self.download_source_file(f"{url}.pdf", created_doc, title)
+
+        self.fetch_relationships(document, created_doc)
+        logger.info("Fetching of relationships is complete!")
+
+    def fetch_relationships(self, imported_document, created_document):
+        subject_work = created_document.work
+
+        if imported_document["repeal"]:
+            repealing_work, _ = Work.objects.get_or_create(
+                frbr_uri=FrbrUri.parse(imported_document["repeal"]["repealing_uri"]),
+                title=imported_document["repeal"]["repealing_title"],
+            )
+            self.create_relationship(
+                predicate_idx=1,
```

Comment:
Magic numbers are always a bit dangerous, since it's unclear if they're correct and easy to go wrong when someone later adds a new one or decides to sort them in the array differently, etc. A more explicit alternative is to use a dict and pass in the predicate slug, rather than a magic number.

```python
self.predicates = {
    # this is implicitly the slug
    "amended-by": {
        "name": "amended by",
        "verb": "is amended by",
        "reverse_verb": "amends",
    },
    ...
}

...

self.create_relationship("amended-by", ...)

...

predicate = self.predicates[slug]
predicate, _ = Predicate.objects.get_or_create(slug, defaults={... slug=slug ...})
```